vault/activity_log.go

```diff
@@ -783,30 +783,29 @@ func (a *ActivityLog) resetCurrentLog() {
 }
 
 func (a *ActivityLog) deleteLogWorker(startTimestamp int64, whenDone chan struct{}) {
-	ctx := namespace.RootContext(nil)
 	entityPath := fmt.Sprintf("%v%v/", activityEntityBasePath, startTimestamp)
 	tokenPath := fmt.Sprintf("%v%v/", activityTokenBasePath, startTimestamp)
 
 	// TODO: handle seal gracefully, if we're still working?
-	entitySegments, err := a.view.List(ctx, entityPath)
+	entitySegments, err := a.view.List(a.core.activeContext, entityPath)
```

Comment:
This is racey since we run without the stateLock held: someone could modify core.activeContext while we're running.

I would instead give deleteLogWorker a ctx argument.  The caller should pass in activeContext, which it reads from the core field while it has the stateLock.  In the case of SetConfig, it's called from a request, so it can assume it has the stateLock.  refreshFromStoredLogs looks like it's called with the stateLock held too.  Same applies to retentionWorker, I think, though you'll have to plumb the ctx through some intervening funcs.

Reply:
f848870fe